src/read/page/reader.rs

```diff
@@ -15,6 +15,28 @@ use crate::page::{
 
 use super::PageIterator;
 
+/// This meta is a small part of [`ColumnChunkMetaData`].
+#[derive(Debug, Clone)]
+pub struct PageMetaData {
+    // The number of values in this column chunk.
+    pub num_values: i64,
+    // Compression type
+    pub compression: Compression,
+    // The descriptor of this parquet column
+    pub descriptor: Descriptor,
+}
+
+impl PageMetaData {
+    /// Returns a new [`PageMetaData`].
+    pub fn new(num_values: i64, compression: Compression, descriptor: Descriptor) -> Self {
+        Self {
+            num_values,
+            compression,
+            descriptor,
+        }
+    }
+}
```

Comment:
I think that we should implement `From<&ColumnChunkMetadata> for PageMetaData` and reimplement `PageReader::new` as `PageReader::new_with_page_meta(reader, column.into(), pages_filter, buffer)` - this is DRY and ensures that this new code path is tested.

What do you think?

Reply:
Agree:)

Reply:
Addressed.

Reply:
Addressed.